website/scripts/setup.sh

```diff
@@ -1,7 +1,28 @@
 #!/bin/bash
 
-brew list hugo || brew install hugo
-which wget || brew install wget
+source ../scripts/detect_os_type.sh
+
+PLATFORM=`platform`
+if [ $PLATFORM = darwin ]; then
+  brew update && brew install nvm && source $(brew --prefix nvm)/nvm.sh
+  nvm install node
+  curl -L https://www.npmjs.com/install.sh | sh 
+  brew list hugo || brew install hugo
+  which wget || brew install wget
+elif [ $PLATFORM = ubuntu ]; then
+  sudo apt-get install golang git mercurial python-pygments -y
+  export GOROOT=/usr/lib/go
+  export GOPATH=$HOME/go
+  export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+  go get -u -v github.com/spf13/hugo
+elif [ $PLATFORM = centos ]; then
+  sudo yum -y install nodejs npm golang --enablerepo=epel
+  export GOROOT=/usr/lib/go
+  export GOPATH=$HOME/go
+  export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+  go get -u -v github.com/spf13/hugo
+fi
+
```

Comment:
@caofangkun - if you install python-pygments above, why we are uninstalling and installing again?